### ISSUE TITLE
Issue with default and keyword arguments

### ISSUE THREAD

**Issue by richardboehme** [seven1m/natalie#206]
If a method has a default argument and keyword arguments, the default argument will be filled with the keyword arguments which is contrary to the behavior of MRI.

```ruby
def foo(a = 1, **)
  puts a 
end

# Natalie:
foo(bar: :foo)
# => { :bar => :foo }

# MRI:
foo(bar: :foo)
# => 1
```

I'm not sure if we have to fix this in `arg_value_by_path` or beforehand in the compiler..

**Comment by richardboehme** [comment]
Also keyword arguments seem to be broken somehow...

```ruby
def foo(**options)
  p "#{options}"
end
foo
# => "[]"
```

Guess that should be an empty Hash 😄

**Comment by seven1m** [comment]
I worked on this a bit, but I have more to do to get this working like MRI does. :disappointed:

**Comment by richardboehme** [comment]
I tried to fix method_with_kwargs9 with ee59029 but I did not come up with a way of fixing it without passing the information that kwargs are present to `arg_value_by_path`. Maybe there is a better way of achieving that?

**Comment by seven1m** [comment]
Yeah, that's where I got stuck too. We need more information passed into the function by the compiler.

**Comment by seven1m** [comment]
Oh, I see; that's what you did. If that works, then yeah, ship it. 😉

**Comment by richardboehme** [comment]
Fixed by #212
